.github/workflows/check.yml

```diff
@@ -1,7 +1,8 @@
 name: "Check with pre-commit"
 on:
-  - "push"
-  - "pull_request"
+  push:
+    branches: ["main"]
+  pull_request:
```

Comment:
These workflows were running twice on pull request changes. This should eliminate the duplicate runs.